addon-resizer/nanny/main/pod_nanny.go

```diff
@@ -84,6 +88,12 @@ func main() {
 	klog.Infof("Version: %s", nanny.AddonResizerVersion)
 	flag.Parse()
 
+	if *exposeMetrics {
+		// Register a handler on the provided address.
+		metrics.Initialize(*address)
+	}
+	metrics.Register()
```

Comment:
Is the order of Initialize and Register relevant? I thought register comes first (does Prometheus do registration at runtime?).
Might be worth having a test to see if the metric is served.